Automattic/pocket-casts-ios
# Player: fix the player getting stuck

We've been receiving some feedback that some users are unable to open the player after a while. While I'm not able to reproduce, I can see why this can happen given we manually control the player state (`open`, `animating`, and `closed`).

Since we changed the player transition to be native (and not custom-controlled) we can use OS variables instead of managing our own.

What I believe is happening is that, for some reason, the player fails to appear but the state is set to `open`. Thus, the player can never be open unless the user closes/reopen the app.

With this change this should be fixed.

## To test

1. Run the app
2. Go to Settings > General > enable "Open Player Automatically"
3. Play any episode
4. ✅ Check that the player appears
5. Dismiss it
6. Tap the miniplayer
7. ✅ It should open again
8. Go to Profile > Settings > Beta Features > disable `newPlayerTransition`
9. Repeat the steps above and make sure they work just fine

## Checklist
